pyV2DL3/eventdisplay/fillRESPONSE.py

```diff
@@ -36,16 +37,75 @@ def print_logging_info(irf_to_store, camera_offsets, pedvar, zenith):
     logging.info(str_woff)
 
 
-def check_parameter_range(par, par_irf, par_name):
-    """Check that coordinates are in range of provided IRF"""
+def check_parameter_range(par, irf_stored_par, par_name):
+    """Check that coordinates are in range of provided IRF and whether extrapolation is to be done
+       0. checks if command line parameter force_extrapolation is given. If given,
+          the extrapolation will happen when parameter is outside IRF range. If parameter is
+          within IRF range, it works as normal. Default is False.
+       1. Further checks for fuzzy boundary (parameter close to boundary value).
+          If fuzzy boundary is within a given tolerance then IRF is interpolated for
+          at boundary value. Default is 0.0 tolerance.
+    """
 
     logging.info(
-        "\t{0} range of a given IRF: {1:.1f} - {2:.1f}".format(
-            par_name, np.min(par_irf), np.max(par_irf)
+        "\t{0} range of a given IRF: {1:.2f} - {2:.2f}".format(
+            par_name, np.min(irf_stored_par), np.max(irf_stored_par)
         )
     )
-    if np.all(par_irf < par) or np.all(par_irf > par):
-        raise ValueError("Coordinate not inside IRF {0} range".format(par_name))
+    clk = click.get_current_context()
+    tolerance = clk.params["fuzzy_boundary"]
+    if np.all(irf_stored_par < par) or np.all(irf_stored_par > par):
+        if clk.params["force_extrapolation"]:
+            logging.warning("IRF extrapolation allowed for coordinate not inside IRF {0} range".format(par_name))
+        else:
+            if tolerance > 0.0:
+                boundary = np.max(irf_stored_par)
```

Comment:
Suggest not the create a variable `boundary` here and in the else step, but put the `np.max(...)` directly in the function call of check_fuzzy_boundary. At this point, the reader sees twice the variable boundary, but it means different things! And you use the variable only once, so no need to define it.